### ISSUE TITLE
Property 'catch' does not exist on type 'PromiseLike '.

### ISSUE THREAD

**Issue by alexisvisco** [timostamm/protobuf-ts#298]
Hello I need to check errors with the catch and I was suprised that UnaryCall implement only PromiseLike, which lack of `catch` method. 

It is maybe related to my tsconfig setup : 

```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noImplicitAny": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "baseUrl": "./src",
    "paths": {
      "@/*": ["*"],
    }
  },
  "include": ["src/app-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}
```

**Comment by timostamm** [comment]
What about this:

```ts
call.then(x => {
   // process response
}).catch(e => {
  // handle call failure - and errors during processing the response
})
```

Or:

```ts
call.then(x => {
   // process response
}, e => {
  // handle call failure - but not errors during processing the response
})
.catch(e => {
  // handle errors during processing the response or handling call failures
})
```

async/await is generally a bit more convenient.

**Comment by alexisvisco** [comment]
1. 
<img width="820" alt="image" src="https://user-images.githubusercontent.com/17145294/164965075-44499fae-736a-47ef-9b2f-4359034e5e74.png">

2.
The second anonymous function in the then seems to work, however the catch did not : 

<img width="766" alt="image" src="https://user-images.githubusercontent.com/17145294/164965270-a488c180-1a18-4fa6-bf89-1d029061f87e.png">

Hope it's more clear for you :) ?

**Comment by alexisvisco** [comment]
This code with the catch works (if i force the compilation), so it is implemented somewhere but the type definition does not correspond to the reality.

**Comment by timostamm** [comment]
Thanks for the details, @alexisvisco. 

The call objects implement `PromiseLike`, because they behave like promises, but aren't actual promises.

In general, it is probably a good idea to use async/await with this API, but I think we can easily use a specific return type `Promise` for our `then()`, while still implementing `PromiseLike`. 

With https://github.com/timostamm/protobuf-ts/pull/308, you should be able to use:

```ts
call.then(x => {
   // process response
}).catch(e => {
  // handle call failure - and errors during processing the response
})
```

**Comment by timostamm** [comment]
Released in [v2.6.0](https://github.com/timostamm/protobuf-ts/releases/tag/v2.6.0)
